servo/api.py

```diff
@@ -120,8 +123,28 @@ def ok(
         return cls(status=ServoStatuses.ok, message=message, reason=reason, **kwargs)
 
     @classmethod
-    def from_error(cls, error: servo.errors.BaseError, **kwargs) -> "Status":
-        """Return a status object representation from the given error."""
+    def from_error(
+        cls, error: servo.errors.BaseError | ExceptionGroup, **kwargs
+    ) -> "Status":
+        """Return a status object representation from the given error (first if multiple in group)."""
+        reason = getattr(error, "reason", ...)
+        if isinstance(error, ExceptionGroup):
+            servo.logger.warning(
+                f"from_error executed on exceptiongroup {error}. May produce undefined behavior"
+            )
+            status = ServoStatuses.failed
+            try:
+                main_err = servo.errors.ServoError.servo_error_from_group(
+                    exception_group=error
+                )
+                if isinstance(main_err, list):
+                    main_err = main_err[0]
+                reason = main_err.reason
+            except:
+                servo.logger.exception(
+                    "Failed to derive exceptiongroup reason for api response"
+                )
+                pass
```

Comment:
> Please use 'except Exception' (or, if at all possible, specify which exception(s) you want to catch).
> remove 'pass'

Will do

> (note, 'reason' will remain set to "...", is this ok for the subsequent cls(..., reason=reason)?

Its preferable for reason to be ... since pydantic treats that as an unset field instead of us setting it to null which would be included on a json dump.